evalml/objectives/regression_objective.py

```diff
@@ -9,5 +9,5 @@ class RegressionObjective(ObjectiveBase):
     problem_type (ProblemTypes): Type of problem this objective is. Set to ProblemTypes.REGRESSION.
     """
 
-    problem_type = ProblemTypes.REGRESSION
+    problem_types = [ProblemTypes.REGRESSION, ProblemTypes.TIME_SERIES_REGRESSION]
```

Comment:
For my own curiosity: Are all regression objectives time-series applicable? Is there ever the case where an objective should be `ProblemTypes.TIME_SERIES_REGRESSION` and not `ProblemTypes.REGRESSION`? :o

Reply:
I would say all of our current regression objectives would also work for a time series problem. But in the future, we will add objectives that are time-series specific, e.g:

https://en.wikipedia.org/wiki/Mean_absolute_scaled_error

Reply:
I think this is the right call.

All our current regression objectives are valid for timeseries regression. So allowing `REGRESSION` and `TIME_SERIES_REGRESSION` is great.

When we add the first time-series-only objective, we can override `problem_types` in each of those impls to only allow `TIME_SERIES_REGRESSION`. We could also then choose to define `TimeseriesRegressionObjective` to facilitate this, but probably not necessary.

We can follow the same pattern for binary/multiclass.